lib/playlistGenerator/PlaylistGenerator.js

```diff
@@ -303,64 +302,75 @@ var persistPlayList = function persistPlayList(force){
             }
         }
     }
-    return that.persistPromise;
+    return that.persistPromise.promise;
 };
 
-var finalizeStart = function (){
+function finalizeStart(){
     var that = this;
-
-
+    // finalizeStart is called only after startPlaylistGenerator is successfully returned.
+    // startPromise is resolved in this function.
+    if (!Q.isPending(that.startPromise)) {
+        return that.startPromise;
+    }
 
     // map flavors onto sequences in correct order
```

Comment:
Do you mean:
if (Q.isResolved(that.startPromise) { return that.startPromise } ?